### PR TITLE
Fix issue when file.check() succeeds when file is closed.

### PR DESCRIPTION
**Pull request by dtjong**
#12373 Seems like hacky solution, but it works. Would be interested to know if there is a better way.

### CODE REVIEW

**Comment by mppf** [comment]
@dtjong - hi, I think it would be better to add a qio_file_isclosed function in runtime/src/qio/qio.h and call that in the .check function. The QIO file struct (in C) has a 'closed' field.

I expect that the solution you have here adds a memory leak.

**Comment by dtjong** [comment]
Hi, I've added qio_file_isopen in qio.h, and the problem is fixed. Under check, should opening a closed file also be lumped in as a `Operation attempted on an invalid file` error? I kept them separate for now.

**Comment by mppf** [comment]
Thanks, this looks good. I'll run testing and get back to you.

- [ ] full local testing

**Comment by mppf** [comment]
@dtjong - could you add a test exercising the new error? It could be based on the code in #12373. Make sure `start_test` passes with your new test. Thanks!

**Comment by mppf** [comment]
hi @dtjong - I'm seeing test failures in `test/library/packages/MatrixMarket/` - you should be able to reproduce with `start_test test/library/packages/MatrixMarket/`. Could you look at them? Thanks.

**Comment by dtjong** [comment]
It seems like it has to do with file.close() calling check, I'll see if I can figure out exactly what's wrong. `start_test` seems to take a really long time to run on my machine, are you doing anything else while testing to speed it up?

**Comment by dtjong** [comment]
I figured it out, the issue is that close is both being explicitly called in `mmwrite()` and called in the deconstructor, closing the file twice. Should it be expected behavior that `close()` throws an error when attempting to close an already closed file, or should it do nothing?

**Comment by mppf** [comment]
> start_test seems to take a really long time to run on my machine, are you doing anything else while testing to speed it up?

You can pass the paths to individual test directiories or .chpl files. I don't recommend you try to run it without doing that (it will run all the tests, which will take probably a half a day or so). We have parallel testing for that.

> Should it be expected behavior that close() throws an error when attempting to close an already closed file, or should it do nothing?

I think doing nothing would be great.

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -1385,8 +1388,6 @@ proc file._style:iostyle throws {
    :throws SystemError: Thrown if the file could not be closed.
  */
 proc file.close() throws {
-  try check();
-
```
Comment (mppf) [inline modules/standard/IO.chpl]: Hmm.. We'd still like the check for is_c_nil to be called. You could copy these lines from file.check and put them here:
``` chapel
  if is_c_nil(_file_internal) then
    throw SystemError.fromSyserr(EBADF, "Operation attempted on an invalid file");
```

**Comment by mppf** [comment]
@dtjong - GitHub is showing a file  test/io/dtjong/foo which should not be included in this PR. Could you fix that please?

**Comment by mppf** [comment]
@dtjong - could you have a look at removing the file that shouldn't be in the PR? Thanks.

**Comment by dtjong** [comment]
Yeah, I'll get on it. Sorry for the late response, have been swamped lately

**Pull request merged**